CHANGELOG.md

```diff
@@ -6,17 +6,28 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
 ## [Unreleased]
+
```

Comment:
Need to rebase and update CHANGELOG

Reply:
Will do after Levi's PR merged in. Any concern for unit test? I'm not sure if I made it too complicated.